src/databricks/labs/ucx/hive_metastore/locations.py

```diff
@@ -293,6 +294,7 @@ def _snapshot(self, fetcher: ResultFn, loader: ResultFn) -> list[Result]:
         cached_results = []
         try:
             cached_results = list(fetcher())
+            self._init_seen_tables(cached_results)
```

Comment:
This is because the crawler can find tables that are already registered in the HMS of the current workspace, which will result in duplicates.
To avoid running into conflicts, it's better to filter tables in mounts that were found during the assessment.